kube-client/src/client/auth/oauth.rs

```diff
@@ -94,15 +100,23 @@ impl Gcp {
             Ok(TokenOrRequest::Request {
                 request, scope_hash, ..
             }) => {
-                #[cfg(not(any(feature = "native-tls", feature = "rustls-tls")))]
+                #[cfg(not(any(feature = "native-tls", feature = "rustls-tls", feature = "openssl-tls")))]
                 compile_error!(
-                    "At least one of native-tls or rustls-tls feature must be enabled to use oauth feature"
+                    "At least one of native-tls or rustls-tls or openssl-tls feature must be enabled to use oauth feature"
                 );
-                // If both are enabled, we use rustls unlike `Client` because there's no need to support ip v4/6 subject matching.
-                // TODO Allow users to choose when both are enabled.
+                // If more than one are enabled, rustls is used unlike `Client` because there's no need to support ip v4/6 subject matching.
+                // - rustls-tls if enabled
+                // - openssl-tls if enabled && !rustls-tls otherwise
+                // - native-tls otherwise
```

Comment:
Yeah, it's intentional, but it doesn't really matter.
```rust
// If more than one are enabled, rustls is used unlike `Client` because there's no need to support ip v4/6 subject matching.
```
I'll change it to match the client to avoid any confusion.